pyfr/backends/openmp/types.py

```diff
@@ -182,7 +182,23 @@ def group(self, kerns, subs=[]):
             groups.append(rargs)
 
         # Arrange for the groupings to be inserted into the final run list
-        self.kins[max(self.knodes[k] for k in kerns) - 1] = groups
+        gdeps = []
+        for k in kerns:
+            gdeps = gdeps + [dep for dep in self.kdeps[k] if dep not in kerns]
+        gid = None
```

Comment:
Why is this needed?  gid is assigned in both branches of the loop.

Reply:
Oops I've very used to pre-declaring variables from other languages sorry